django/http/request.py

```diff
@@ -480,15 +480,22 @@ def build_request_repr(request, path_override=None, GET_override=None,
                else pformat(request.GET))
     except Exception:
         get = '<could not parse>'
+
     if request._post_parse_error:
         post = '<could not parse>'
     else:
         try:
-            post = (pformat(POST_override)
-                    if POST_override is not None
-                    else pformat(request.POST))
+            if POST_override is not None:
+                post = pformat(POST_override)
+            else:
+                # Use the exception reporter filter to return the cleansed POST data
+                from django.views.debug import get_exception_reporter_filter
```

Comment:
POST data shouldn't leak anymore, so the ticket object is solved. But I would keep the refactoring of the `view_tests` because current test are too artifical so they can hide some sensitive data problems. They only produce tracebacks with single frame which is very unusual in real application. If needed I may create another ticket to refactor the tests.